src/realm/table.cpp

```diff
@@ -3257,8 +3257,6 @@ void Table::set_binary(size_t col_ndx, size_t ndx, BinaryData value, bool is_def
         throw LogicError(LogicError::column_index_out_of_range);
     if (!is_nullable(col_ndx) && value.is_null())
         throw LogicError(LogicError::column_not_nullable);
-    if (REALM_UNLIKELY(value.size() > ArrayBlob::max_binary_size))
-        throw LogicError(LogicError::binary_too_big);
```

Comment:
This interface will be used for storing a transaction log in the sync database, so this will probably not be synced to another db, I will discuss it anyway.